components/gitpod-db/src/typeorm/user-db-impl.ts

```diff
@@ -395,6 +402,43 @@ export class TypeORMUserDBImpl implements UserDB {
         await repo.save(envVar);
     }
 
+    public async hasSSHPublicKey(userId: string): Promise<boolean> {
+        const repo = await this.getSSHPublicKeyRepo();
+        return !!(await repo.findOne({ where: { userId, deleted: false } }));
+    }
+
+    public async getSSHPublicKeys(userId: string): Promise<UserSSHPublicKey[]> {
+        const repo = await this.getSSHPublicKeyRepo();
+        return repo.find({ where: { userId, deleted: false }, order: { creationTime: "ASC" } });
```

Comment:
Note that this query is not backed 100% by an index. It should still work, because:
 1. userId is indexed, and has the highest cardinality, so the query engine should go for this first
 2. creationtime has a low cardinality
 3. deleted has a low cardinality

Reply:
Is it means we should refer an index here to find?

Reply:
I think it's fine as-is, just wanted to share my thoughts. :+1: